src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/TrustHelper.cs

```diff
@@ -947,14 +947,8 @@ private static void ValidateTrustAttribute(TRUSTED_DOMAIN_INFORMATION_EX domainI
 
         internal static string CreateTrustPassword()
         {
-#if NETCOREAPP3_0_OR_GREATER
-            return string.Create<object?>(PASSWORD_LENGTH, null, static (destination, _) =>
-            {
-                for (int i = 0; i < destination.Length; i++)
-                {
-                    destination[i] = PasswordCharacterSet[RandomNumberGenerator.GetInt32(PasswordCharacterSet.Length)];
-                }
-            });
+#if NET8_0_OR_GREATER
+            return RandomNumberGenerator.GetString(PasswordCharacterSet, PASSWORD_LENGTH);
```

Comment:
Note that this will end up removing the optimization that was there using string.Create when building any downlevel .NET Core targets.

I'm not actually concerned about it in this case, nor do I think it's worth adding another copy of the code under `#else if`, just wanted to call it out.

Reply:
Yeah, that was intentional. Thanks for pointing it out though.